archunit/src/main/java/com/tngtech/archunit/core/domain/JavaClass.java

```diff
@@ -1342,7 +1358,7 @@ private Superclass(Optional<JavaType> type) {
         }
 
         Set<JavaClass> getAllInterfaces() {
-            return rawType.isPresent() ? rawType.get().getAllInterfaces() : Collections.<JavaClass>emptySet();
+            return rawType.isPresent() ? rawType.get().getAllRawInterfaces() : Collections.<JavaClass>emptySet();
         }
 
         Superclass withRawType(JavaClass newRawType) {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Class names should start with an upper case letter (NM_CLASS_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-class-names-should-start-with-an-upper-case-letter-nm-class-naming-convention).
Class names should be nouns, in mixed case with the first letter of each internal word capitalized. Try to keep your class names simple and descriptive. Use whole words-avoid acronyms and abbreviations (unless the abbreviation is much more widely used than the long form, such as URL or HTML).

Reply:
Are you talking about the camel casing of `Superclass`? That was actually a longer thought process :wink: You can compare this to the Reflection API: `Class.getSuperclass()`, so this is where it is coming from (besides that there is also "Superman", "supermarket" and "superconductor", i.e. "super" is in general not considered a separate word for alle examples I could find in the English language)

Reply:
If you're not talking about that, then please elaborate a little more precisely where exactly you think the problem is.

Reply:
I wonder where some automatic pattern matching has detected `class withRawType`... 😝

Reply:
My mistake. It is a method function instead of a class.